tests/test_timing.py

```diff
@@ -511,33 +510,30 @@ def test_logged_duration_real_logger(
     # sequence of log dicts to expect to be output as json logs
     expected_logs,
 ):
-    with open(app_logtofile.config["DM_LOG_PATH"], "r") as log_file:
-        # consume log initialization line
-        log_file.read()
-
-        with app_logtofile.test_request_context("/", headers={}):
-            request.is_sampled = is_sampled
-
-            with (mock_time_functions() if mock_time else actual_time_functions()) as (
-                _sleep,
-                _perf_counter,
-                _process_time,
-            ):
-                with (null_context_manager() if raise_exception is None else pytest.raises(raise_exception)):
-                    with timing.logged_duration(
-                        logger=app_logtofile.logger.getChild("foobar"),
-                        message=message,
-                        log_level=log_level,
-                        condition=condition,
-                    ) as log_context:
-                        _sleep(sleep_time)
-                        if inject_context is not None:
-                            log_context.update(inject_context)
-                        if raise_exception is not None:
-                            raise raise_exception("Boo")
-
-        # ensure buffers are flushed
-        logging.shutdown()
-
-        all_lines = tuple(json.loads(line) for line in log_file.read().splitlines())
-        assert all_lines == expected_logs
+    app, stream = app_with_stream_logger
+
+    with app.test_request_context("/", headers={}):
+        request.is_sampled = is_sampled
+
+        with (mock_time_functions() if mock_time else actual_time_functions()) as (
+            _sleep,
+            _perf_counter,
+            _process_time,
+        ):
+            with (null_context_manager() if raise_exception is None else pytest.raises(raise_exception)):
+                with timing.logged_duration(
+                    logger=app.logger.getChild("foobar"),
+                    message=message,
+                    log_level=log_level,
+                    condition=condition,
+                ) as log_context:
+                    _sleep(sleep_time)
+                    if inject_context is not None:
+                        log_context.update(inject_context)
+                    if raise_exception is not None:
+                        raise raise_exception("Boo")
+
+    stream.seek(0)
```

Comment:
Though it doesn't directly matter here, I think it's probably a better habit to use `StringIO`'s `.getvalue()` method for this rather than reaching in and seeking around the file. If someone were to make an assertion like this and leave the (imaginary) file descriptor not at the end of the file & continue on with a couple of extra steps of the test, you'd get some confusing mixed up output and it wouldn't be clear why.

Reply:
Yeah I tried that but it meant that I had to mess with the comprehension below quite a lot and I wanted to leave it as similar to before as possible. `get_value` returns the whole string joined by linebreaks 😒

Reply:
:confused: Isn't that exactly what `stream.read()` does?

Reply:
Good call, can use getvalue and remove the seek line 💯